dali/operators/geometry/affine_transforms/crop_transform.cc

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/geometry/affine_transforms/transform_base_op.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+DALI_SCHEMA(CropTransform)
+  .DocStr(R"code(Produces an affine transform matrix that maps a reference to another one.
+
+This transform can be used to adjust coordinates after a crop operation so that a ``from_start`` point will
+be mapped to ``to_start`` and ``from_end`` will be mapped to ``to_end``.
+
+If another transform matrix is passed as an input, the operator applies the transformation to the matrix provided.
+
+.. note::
+    The output of this operator can be fed directly to the ``MT`` argument of ``CoordTransform`` operator.
+)code")
+  .AddArg(
+    "from_start",
+    R"code(The lower bound of the original coordinate space.)code",
+    DALI_FLOAT_VEC, true)
+  .AddArg(
+    "from_end",
+    R"code(The upper bound of the original coordinate space.)code",
+    DALI_FLOAT_VEC, true)
+  .AddArg(
+    "to_start",
+    R"code(The lower bound of the destination coordinate space.)code",
+    DALI_FLOAT_VEC, true)
+  .AddArg(
+    "to_end",
+    R"code(The upper bound of the destination coordinate space.)code",
+    DALI_FLOAT_VEC, true)
+  .AddOptionalArg(
+    "absolute",
+    R"code(If set to true, start and end coordinates will be swapped if start > end.)code",
+    false, true)
+  .NumInput(0, 1)
+  .NumOutput(1)
+  .AddParent("TransformAttr");
+
+/**
+ * @brief Scale transformation.
+ */
+class CropTransformCPU
+    : public TransformBaseOp<CPUBackend, CropTransformCPU> {
+ public:
+  using SupportedDims = dims<1, 2, 3, 4, 5, 6>;
+
+  explicit CropTransformCPU(const OpSpec &spec) :
+      TransformBaseOp<CPUBackend, CropTransformCPU>(spec),
+      from_start_("from_start", spec),
+      from_end_("from_end", spec),
+      to_start_("to_start", spec),
+      to_end_("to_end", spec) {
+  }
+
+  template <typename T, int mat_dim>
+  void DefineTransforms(span<affine_mat_t<T, mat_dim>> matrices) {
+    constexpr int ndim = mat_dim - 1;
+    assert(matrices.size() == static_cast<int>(from_start_.size()));
```

Comment:
Assert or Enforce?

Reply:
You can just enforce one arg, the equality of the rest is checked in ProcessArgs./=

Reply:
assert, just a sanity check